test/standalone/windows_fs/lock.zig

```diff
@@ -0,0 +1,22 @@
+const std = @import("std");
+pub fn main() !void {
+    std.fs.cwd().deleteFile("lockfile") catch |err| switch (err) {
+        error.FileNotFound => {},
+        else => |e| return e,
+    };
+    {
+        const f = try std.fs.cwd().createFile("lockfile", .{ .read = true });
+        defer f.close();
+        var buffer: [1]u8 = undefined;
+        _ = try f.readAll(&buffer);
+    }
+    {
+        const f = try std.fs.cwd().createFile("lockfile", .{
+            .read = true,
+            .lock = .exclusive,
+        });
+        var buffer: [1]u8 = undefined;
+        try std.testing.expectError(error.LockViolation, f.readAll(&buffer));
+        try std.fs.cwd().deleteFile("lockfile");
+    }
+}
```

Comment:
Is there are reason this needs to be a standalone test instead in [`fs/test.zig`](https://github.com/ziglang/zig/blob/master/lib/std/fs/test.zig)?